src/pages/Admin/AdminMainLayout.tsx

```diff
@@ -1,25 +1,50 @@
-import React from 'react';
-import { Outlet } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import { Outlet, useNavigate, useOutletContext } from 'react-router-dom';
 import styled from 'styled-components';
+import { getAdminCohorts } from '../../api/getAdminCohorts';
+import { Cohort, ErrorMessageType } from '../../types';
+import { ERROR_MESSAGES } from '../../utilities/constants';
 import AdminNavbar from './AdminNavbar';
 
 const Container = styled.div`
   width: 100vw;
   height: 100vh;
-  padding: 0 60px;
   display: flex;
   background: ${({ theme }) => theme.background.secondary}; ;
 `;
 
-const AdminMainLayout = () => {
-  return (
+type AdminContextType = { cohorts: Cohort[] | null };
+
+export const AdminMainLayout = () => {
+  const [cohortsList, setCohortsList] = useState<Cohort[]>([]);
+  const navigate = useNavigate();
+
+  const fetchAdminCohorts = async () => {
+    try {
+      const res = await getAdminCohorts();
+      setCohortsList(res.data);
+    } catch (error) {
+      const errors = error as ErrorMessageType[];
+      if (errors?.[0].msg === ERROR_MESSAGES.unauthorized) {
+        navigate('/sign-in', { replace: true });
+      }
+    }
+  };
+
+  useEffect(() => {
+    fetchAdminCohorts();
+  });
+
+  return cohortsList.length ? (
     <div>
       <Container>
         <AdminNavbar />
-        <Outlet />
+        <Outlet context={{ cohorts: cohortsList }} />
       </Container>
     </div>
-  );
+  ) : null;
 };
```

Comment:
the main idea here is that we try to call GET `/admin/cohorts` to see if the user is authorized or not, if they are, the backend will return `cohorts`, otherwise `401` and we redirect the page to `/sign-in`.

So in here, we just pass down the `cohorts` as props to the `Outlet` context, which means any pages under `Outlet` will have the `cohorts` list in the global context.

Reply:
I personally prefer to just passing authenticated state/not than passing the whole cohorts array. Then fetch it later when the user opens up the page through GET request.